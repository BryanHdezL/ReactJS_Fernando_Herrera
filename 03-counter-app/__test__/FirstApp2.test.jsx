const { render, screen } = require("@testing-library/react");
const { FirstApp } = require("../src/FirstApp");
import React from 'react'

describe('Pruebas en <FirstApp />', () => {

    const title = 'Bryan Hernandez'
    const number = 22

    test('Debe de hacer match con el snapshot', () => {
        const { container } = render(<FirstApp title={title} number={number} />)
        expect(container).toMatchSnapshot();
    });


    test('Debe de mostrar el titulo', () => {
        render(<FirstApp title={title} number={number} />)
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('Debe de mostrar el titulo en un H1', () => {
        render(<FirstApp title={title} number={number} />)
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

});
