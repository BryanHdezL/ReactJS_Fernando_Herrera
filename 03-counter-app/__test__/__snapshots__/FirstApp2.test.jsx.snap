// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Bryan Hernandez
  </h1>
  <h2>
    Tengo 
    22
     a√±os
  </h2>
</div>
`;
